python_modules/dagster/dagster/_core/definitions/asset_condition.py

```diff
@@ -77,58 +77,100 @@ def get_serializable_candidate_subset(
-        return unpack_value(
-            serialized_candidate_subset,
-            (AssetSubset, HistoricalAllPartitionsSubsetSentinel),
-            whitelist_map,
-            context,
+    @staticmethod
+    def create(
+        context: "AssetConditionEvaluationContext",
+        true_subset: AssetSubset,
+        subsets_with_metadata: Sequence[AssetSubsetWithMetadata] = [],
+        extra_state: PackableValue = None,
+    ) -> "AssetConditionResult":
+        """Returns a new AssetConditionEvaluation from the given parameters."""
+        return AssetConditionResult(
+            condition=context.condition,
+            start_timestamp=context.start_timestamp,
+            end_timestamp=pendulum.now("UTC").timestamp(),
+            true_subset=true_subset,
+            candidate_subset=context.candidate_subset,
+            subsets_with_metadata=subsets_with_metadata,
+            child_results=[],
+            extra_state=extra_state,
         )
 
 
-@whitelist_for_serdes(field_serializers={"candidate_subset": CandidateSubsetSerializer})
+@whitelist_for_serdes
 class AssetConditionEvaluation(NamedTuple):
-    """Internal representation of the results of evaluating a node in the evaluation tree."""
+    """Serializable representation of the results of evaluating a node in the evaluation tree."""
 
     condition_snapshot: AssetConditionSnapshot
-    true_subset: AssetSubset
-    candidate_subset: Union[AssetSubset, HistoricalAllPartitionsSubsetSentinel]
     start_timestamp: Optional[float]
     end_timestamp: Optional[float]
-    subsets_with_metadata: Sequence[AssetSubsetWithMetadata] = []
-    child_evaluations: Sequence["AssetConditionEvaluation"] = []
+
+    true_subset: AssetSubset
+    candidate_subset: Union[AssetSubset, HistoricalAllPartitionsSubsetSentinel]
+    subsets_with_metadata: Sequence[AssetSubsetWithMetadata]
+
+    child_evaluations: Sequence["AssetConditionEvaluation"]
 
     @property
     def asset_key(self) -> AssetKey:
         return self.true_subset.asset_key
 
+    @staticmethod
+    def from_result(result: AssetConditionResult) -> "AssetConditionEvaluation":
+        return AssetConditionEvaluation(
+            condition_snapshot=result.condition.snapshot,
+            start_timestamp=result.start_timestamp,
+            end_timestamp=result.end_timestamp,
+            true_subset=result.true_subset,
+            candidate_subset=get_serializable_candidate_subset(result.candidate_subset),
+            subsets_with_metadata=result.subsets_with_metadata,
+            child_evaluations=[
+                AssetConditionEvaluation.from_result(child_result)
+                for child_result in result.child_results
+            ],
+        )
+
     def equivalent_to_stored_evaluation(self, other: Optional["AssetConditionEvaluation"]) -> bool:
-        """Returns if all fields other than `run_ids` are equal."""
+        """Returns if this evaluation is functionally equivalent to the given stored evaluation.
```

Comment:
What are the situations where they would be considered functionality equivalent but not `__equal__`? Worth mentioning at least an example here.

Reply:
added